packages/circuits/circuits/bigComparators.circom

```diff
@@ -5,57 +5,50 @@ include "./modulo.circom";
 
 //~~ support comparisons of numbers up to the field size
 
-template UpperLessThan(n) {
-    assert(n < 254);
+template LowerLessThan(n) {
+    assert(n < 253);
     signal input in[2];
     signal output out;
 
-    component bits[2];
+    signal upper_values[2];
+    signal lower_values[2];
+    // to satisfy compiler
+    signal _v[2][n];
     for (var x = 0; x < 2; x++) {
-        bits[x] = Num2Bits_strict();
-        bits[x].in <== in[x];
-    }
-
-    component upper_bits[2];
-    upper_bits[0] = Bits2Num(n);
-    upper_bits[1] = Bits2Num(n);
-
-    for (var x = 0; x < n; x++) {
-        upper_bits[0].in[x] <== bits[0].out[x+(254-n)];
-        upper_bits[1].in[x] <== bits[1].out[x+(254-n)];
+        upper_values[x] <-- in[x] >> n;
+        // lower_values[x] should contain the lower bits
+        lower_values[x] <== in[x] - upper_values[x] * 2**n;
+        // verify that lower_values[x] does not have more than lower bits
+        _v[x] <== Num2Bits(n)(lower_values[x]);
```

Comment:
will `assert(n < 254);` be wrong here?

Reply:
Yes, it must be `< 253` because we use a `LessThan` component which has a max of `252` bits